### PR TITLE
machine/nrf: refactor to use volatile package/API

### PR DESCRIPTION
**Pull request by deadprogram**
This PR refactors the NRF boards to use the volatile package/API. It needs to have at least issue #363 addressed before it can be merged.

### CODE REVIEW

**Comment by deadprogram** [comment]
All tests are now passing, I think this PR is ready to merge.

File: src/machine/machine_nrf.go [hunk 1]
```diff
@@ -20,38 +20,38 @@ const (
...
+		port.OUTCLR.Set(1 << pin)
 	}
 }
 
 // Return the register and mask to enable a given GPIO pin. This can be used to
 // implement bit-banged drivers.
 func (p GPIO) PortMaskSet() (*uint32, uint32) {
 	port, pin := p.getPortPin()
-	return (*uint32)(&port.OUTSET), 1 << pin
+	return (*uint32)(&port.OUTSET.Reg), 1 << pin
```
Comment (aykevl) [inline src/machine/machine_nrf.go]: This cast (and the one below) is unnecessary.

Comment (deadprogram) [reply src/machine/machine_nrf.go]: OK fixed.

**Comment by aykevl** [comment]
:confetti_ball:

**Pull request merged**